### PR TITLE
feat(study): add review streak tracking and home indicator

### PR DESCRIPTION
**Pull request by DevamPatel22**
Adds review streak tracking from completed study sessions and shows a live streak indicator on the home screen while keeping streak stats updated for current, longest, and last completed progress.

### CODE REVIEW

**Review by Pairadux — COMMENTED** [review]
## Review

Looks good overall — nice work wiring up session summary persistence and the streak UI. Tests are solid with good edge case coverage, and the `asOf` parameter for testability is a nice touch.

### Stale Streak Data

`reviewStreakProvider` is a plain `FutureProvider` that caches until explicitly invalidated. If the app stays open past midnight the streak won't refresh. Either switch to `FutureProvider.autoDispose` or invalidate on app lifecycle resume via `WidgetsBindingObserver.didChangeAppLifecycleState`.

### getStreak Performance

`getStreak()` loads every distinct date into Dart and computes in memory. Fine for now, but the current streak could be a backwards SQL scan from today/yesterday and longest streak could use a window function. Worth a follow-up.

File: lib/features/study/data/review_session_summary_repository.dart [hunk 1]
```diff
@@ -115,4 +116,118 @@ class ReviewSessionSummaryRepository {
...
+          )
+        )
+        SELECT COUNT(*) AS current_streak
+        FROM streak
+        WHERE EXISTS (
+          SELECT 1
+          FROM ${DatabaseConstants.tableReviewSessionSummary}
+          WHERE ${DatabaseConstants.colDate} = streak.day
+            AND ${DatabaseConstants.colTotalReviews} > 0
+        )
```
Comment (Pairadux) [inline lib/features/study/data/review_session_summary_repository.dart]: Minor: the outer `WHERE EXISTS` here never filters anything. Every row in the `streak` recursive CTE is added because the prior day's row exists, and the base case is the anchor day, which is guaranteed to exist by the `has_today`/`has_yesterday` check above. Safe to drop:

```suggestion
        SELECT COUNT(*) AS current_streak
        FROM streak
```

File: lib/features/study/presentation/screens/review_stats_screen.dart [hunk 2]
```diff
@@ -100,7 +103,91 @@ class _ReviewStatsScreenState extends ConsumerState<ReviewStatsScreen> {
...
+            ).textTheme.labelMedium?.copyWith(color: AppColors.textTertiary),
+          ),
+          const SizedBox(height: Spacing.xs),
+          Text(
+            value,
+            style: Theme.of(context).textTheme.titleLarge?.copyWith(
+              color: color,
+              fontWeight: FontWeight.w700,
+            ),
+          ),
```
Comment (Pairadux) [inline lib/features/study/presentation/screens/review_stats_screen.dart:187]: On narrow phones, a `YYYY-MM-DD` date in `titleLarge` next to two other Expanded cards can overflow. Guarding the Text avoids a layout exception:

```suggestion
          Text(
            value,
            maxLines: 1,
            overflow: TextOverflow.ellipsis,
            style: Theme.of(context).textTheme.titleLarge?.copyWith(
              color: color,
              fontWeight: FontWeight.w700,
            ),
          ),
```

A friendlier format (e.g. `Mar 19`) would also be nice but is a separate change.

**Review by Pairadux — COMMENTED** [review]
Both prior comments addressed thoroughly — `autoDispose` + day/resume refresh providers, and SQL-based streak computation with gap-and-islands. Tests cover all the right edges.

Left two nitpicks below as suggestions you can apply directly. The flame animation can stay as-is.
